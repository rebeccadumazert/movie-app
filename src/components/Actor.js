import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import {
  callDetailCredit,
  callDetailCreditMovie,
} from './../services/moviesDb';
import './style/actor.css';

export default class Actor extends Component {
  state = {
    actor: {},
    filmography: [],
  };

  componentDidMount() {
    this.fetchActorData();
  }
  componentDidUpdate(prevProps) {
    if (this.props.match.params.idCredit !== prevProps.match.params.idCredit) {
      this.fetchActorData();
    }
  }

  fetchActorData = () => {
    this.fetchDetailsActor();
    this.fetchCreditMovie();
  };

  async fetchDetailsActor() {
    const {
      match: { params },
    } = this.props;
    const { data: actor } = await callDetailCredit(params.idCredit);
    this.setState({
      actor,
    });
  }
  async fetchCreditMovie() {
    const {
      match: { params },
    } = this.props;
    const {
      data: { cast: filmography },
    } = await callDetailCreditMovie(params.idCredit);
    this.setState({
      filmography,
    });
  }
  render() {
    const {
      actor: {
        name,
        homepage,
        biography,
        birthday,
        place_of_birth,
        profile_path,
      },
      filmography,
    } = this.state;
    return (
      <div>
        <h1 className="titleActor">{name}</h1>
        <div className="containerImgActor">
          <img
            className="imgActor"
            src={
              !profile_path
                ? `http://cul7ure.fr/wp-content/uploads/2017/09/1216-1024x1024.jpg`
                : `https://image.tmdb.org/t/p/w500/${profile_path}`
            }
            alt="actor"
          />

          <div className="infoActorContainer">
            <p>Lieu de naissance : {place_of_birth}</p>
            <p>Anniversaire : {birthday}</p>
            <p>Biographie : {biography}</p>
            {!!homepage && (
              <p>
                Site Web : <a href={homepage}>{homepage}</a>
              </p>
            )}
            <div>
              Filmographie :{' '}
              {filmography.map(movie => (
                <Link key={movie.id} to={`/movie/${movie.id}`}>
                  <p>{movie.title}</p>
                </Link>
              ))}
            </div>
          </div>
        </div>
      </div>
    );
  }
}
