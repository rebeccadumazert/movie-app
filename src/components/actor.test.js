import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Actor from './Actor';

configure({ adapter: new Adapter() });

test('Actor renders correctly', () => {
  const tree = renderer.create(<Actor />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('component should contain anniversaire', () => {
  const actor = shallow(<Actor />);
  expect(actor.text()).toMatch(/Anniversaire/);
});

test('component should contain h1', () => {
  const actor = shallow(<Actor></Actor>);
  expect(actor.find('h1')).toHaveLength(1);
});

// test('')
