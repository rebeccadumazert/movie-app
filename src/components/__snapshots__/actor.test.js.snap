// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actor renders correctly 1`] = `
<div>
  <h1
    className="titleActor"
  />
  <div
    className="containerImgActor"
  >
    <img
      alt="pix actor/actress"
      className="imgActor"
      src="http://cul7ure.fr/wp-content/uploads/2017/09/1216-1024x1024.jpg"
    />
    <div
      className="infoActorContainer"
    >
      <p>
        Lieu de naissance : 
      </p>
      <p>
        Anniversaire : 
      </p>
      <p>
        Biographie : 
      </p>
      Filmographie :
      <div
        className="filmoContainer"
      />
    </div>
  </div>
</div>
`;
